x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/TransactionTabs.tsx

```diff
@@ -76,16 +64,77 @@ export function TransactionTabs({
 
       <EuiSpacer />
 
-      {currentTab.key === timelineTab.key ? (
-        <WaterfallContainer
-          location={location}
-          urlParams={urlParams}
-          waterfall={waterfall}
-          exceedsMax={exceedsMax}
-        />
-      ) : (
-        <TransactionMetadata transaction={transaction} />
-      )}
+      <TabContent
+        location={location}
+        urlParams={urlParams}
+        waterfall={waterfall}
+        exceedsMax={exceedsMax}
+        transaction={transaction}
+      />
     </React.Fragment>
   );
 }
+
+const timelineTab = {
+  key: 'timeline',
+  label: i18n.translate('xpack.apm.propertiesTable.tabs.timelineLabel', {
+    defaultMessage: 'Timeline',
+  }),
+  component: TimelineTabContent,
+};
+
+const metadataTab = {
+  key: 'metadata',
+  label: i18n.translate('xpack.apm.propertiesTable.tabs.metadataLabel', {
+    defaultMessage: 'Metadata',
+  }),
+  component: MetadataTabContent,
+};
+
+const logsTab = {
+  key: 'logs',
+  label: i18n.translate('xpack.apm.propertiesTable.tabs.logsLabel', {
+    defaultMessage: 'Logs',
+  }),
+  component: LogsTabContent,
+};
+
+function TimelineTabContent({
+  location,
+  urlParams,
+  waterfall,
+  exceedsMax,
+}: {
+  location: Location<any>;
+  urlParams: IUrlParams;
+  waterfall: IWaterfall;
+  exceedsMax: boolean;
+}) {
+  return (
+    <WaterfallContainer
+      location={location}
+      urlParams={urlParams}
+      waterfall={waterfall}
+      exceedsMax={exceedsMax}
+    />
+  );
+}
+
+function MetadataTabContent({ transaction }: { transaction: Transaction }) {
+  return <TransactionMetadata transaction={transaction} />;
+}
+
+function LogsTabContent({ transaction }: { transaction: Transaction }) {
+  const startTimestamp = Math.floor(transaction.timestamp.us / 1000);
+  const endTimestamp = Math.ceil(
+    startTimestamp + transaction.transaction.duration.us / 1000
+  );
+  const framePaddingMs = 1000 * 60 * 60 * 24; // 24 hours
+  return (
+    <LogStream
+      startTimestamp={startTimestamp - framePaddingMs}
+      endTimestamp={endTimestamp + framePaddingMs}
+      query={`trace.id:"${transaction.trace.id}" OR "${transaction.trace.id}"`}
```

Comment:
This matches the filter for the "Trace logs" link in the action menu: https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts#L183, but i'm not sure why this needs to be an `OR`, is it just me or is this redundant?

Reply:
My guess is that is will pick up entries that don't have the trace.id attribute set but contain the trace id in other fields, but first matching with the field name gives exact matches.